extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/AvroOCFInputFormat.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.avro;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.avro.Schema;
+import org.apache.druid.data.input.InputEntity;
+import org.apache.druid.data.input.InputEntityReader;
+import org.apache.druid.data.input.InputRowSchema;
+import org.apache.druid.data.input.impl.NestedInputFormat;
+import org.apache.druid.guice.annotations.Json;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.common.parsers.JSONPathSpec;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Map;
+import java.util.Objects;
+
+public class AvroOCFInputFormat extends NestedInputFormat
+{
+  private static final Logger LOGGER = new Logger(AvroOCFInputFormat.class);
+
+  private final boolean binaryAsString;
+  private final Schema readerSchema;
+
+  @JsonCreator
+  public AvroOCFInputFormat(
+      @JacksonInject @Json ObjectMapper mapper,
+      @JsonProperty("flattenSpec") @Nullable JSONPathSpec flattenSpec,
+      @JsonProperty("schema") Map<String, Object> schema,
+      @JsonProperty("binaryAsString") @Nullable Boolean binaryAsString
+  ) throws Exception
+  {
+    super(flattenSpec);
+    // If a reader schema is supplied create the datum reader with said schema, otherwise use the writer schema
+    if (schema != null) {
+      String schemaStr = mapper.writeValueAsString(schema);
+      LOGGER.debug("Initialising with reader schema: [%s]", schemaStr);
+      this.readerSchema = new Schema.Parser().parse(schemaStr);
+    } else {
+      this.readerSchema = null;
+    }
+    this.binaryAsString = binaryAsString == null ? false : binaryAsString;
+  }
+
+  @Override
+  public boolean isSplittable()
+  {
+    return false;
```

Comment:
This seems fine for now, but we would probably want to make it splittable in the future. Would you please add a comment about it with this [link](https://avro.apache.org/docs/current/spec.html#Object+Container+Files)?

Reply:
Yeah I think the best way to do splitting would be block-wise. I will add a comment about it and link to the spec as you suggested.